langchain/document_loaders/text.py

```diff
@@ -8,13 +7,14 @@
 class TextLoader(BaseLoader):
     """Load text files."""
 
-    def __init__(self, file_path: str):
+    def __init__(self, file_path: str, encoding: Optional[str] = None):
         """Initialize with file path."""
         self.file_path = file_path
+        self.encoding = encoding
 
     def load(self) -> List[Document]:
         """Load from file path."""
-        with open(self.file_path, encoding="utf-8") as f:
+        with open(self.file_path, encoding=self.encoding) as f:
```

Comment:
Addition of 'r' parameter fixes the `UnicodeDecodeError`. Could you also add that acc. to #2223

Reply:
From first glance, I believe mode='r' is the default, so I do not know why that would cause a problem. https://docs.python.org/3/library/functions.html#open

Reply:
That is true. The updated version 0.0.129 did fix that issue. I am not sure why I had that issue in Windows. btw, would it be better if the default encoding is 'utf-8' as opposed to None.